dotnet/src/SemanticKernel/CoreSkills/PlannerSkill.cs

```diff
@@ -242,19 +260,96 @@ public async Task<SKContext> CreatePlanAsync(string goal, SKContext context)
         context.Variables.Set("available_functions", relevantFunctionsManual);
         // TODO - consider adding the relevancy score for functions added to manual
 
-        var plan = await this._functionFlowFunction.InvokeAsync(context);
+        if (await this.VerifyCanCreatePlanForGoalAsync(goal, context))
```

Comment:
I don't think we should hit the model before creating a plan, just to ask if it can. Extra time, expense. This should be merged into the Planner prompt itself: "Create a plan. If you can't generate a plan, respond with..."

Reply:
I think there are some other ways we can do this, let's connect.